src/js/controllers/create.js

```diff
@@ -53,23 +59,50 @@ angular.module('copayApp.controllers').controller('createController',
         name: form.walletName.$modelValue,
         extendedPrivateKey: form.privateKey.$modelValue,
         myName: $scope.totalCopayers > 1 ? form.myName.$modelValue : null,
-        networkName: form.isTestnet.$modelValue ? 'testnet' : 'livenet'
+        networkName: form.isTestnet.$modelValue ? 'testnet' : 'livenet',
       };
       self.loading = true;
 
+      if (form.hwLedger.$modelValue) {
+        self.ledger = true;
+        ledger.getXPubKey($scope.externalIndex, function(data) {
+          self.ledger = false;
+          $scope.$apply();
+          if (data.success) {
+            opts.extendedPublicKey = data.xpubkey;
+            opts.externalSource = 'ledger';
+            opts.externalIndex = $scope.externalIndex;
+            self._create(opts);
+          } else {
+            self.loading = false;
+            $log.debug(data.message);
+            self.error = data.message;
+            $scope.$apply();
```

Comment:
it seemsthat $scope.$apply will be called 2 times if data.sucess is false. This could lead to an angular error.